sycl/source/detail/pi.cpp

```diff
@@ -447,6 +447,8 @@ static void initializePlugins(std::vector<plugin> *Plugins) {
   }
 
 #ifdef XPTI_ENABLE_INSTRUMENTATION
+  GlobalHandler::instance().getXPTIRegistry().initializeFrameworkOnce();
```

Comment:
Do we really need to initialize XPTI if it is disabled (see check at line 452)?

Reply:
Yes. Whether tracing is enabled or disabled is decided by the dispatcher library. SYCL runtime is linked with so-called proxy library, which only forwards calls to the dispatcher. As a result that method returns false in two cases: a) we failed to initialize dispatcher library; b) the dispatcher library returned false for any reason. So, yes, we need to initialize XPTI in order to figure out whether tracing is enabled or not. The only thing this patch changes is addition of a new init/deinit methods that allow us to bypass the whole global c'tor/d'tor invokation order problem and manually manage our global state.